docs/nsm_api.md

```diff
@@ -0,0 +1,445 @@
+    </dd>
+    <dt>Network Service Endpoint (NSE)</dt>
+    <dd>
+        A Network Service Endpoint provides a concrete usable instance of a Network Service for consumption by Network Service Clients.  Example: Pods providing a Network Service in Kubernetes.
+    </dd>
+</dl>
+
+![Network Service Mesh Abstract Components](./images/NSM%20Domains.png)
+<dl>
+    <dt>Network Service Dataplane (NSD)</dt>
+    <dd>
+        Within the Network Service Manager domain, the Network Service Dataplan is the dataplane managed by the Network Service Manager to 
+        <ul>
+            <li>Connect Network Service Clients in its domain to Network Service Endpoints.  The Network Service Endpoints need not be in the Network Service Managers domain.</li>
+            <li>Connect Network Service Endpoints in its domain to Network Service Clients.  The Network Service Clients need not be in the Network Service Managers domain, nor within any Service Registry Domain that the Network Service Manager participates in.</li>
+        <ul>
+    </dd>
+    <dt>Network Service Manager (NSM)</dt>
+    <dd>
+        A Network Service Manager (NSM) manages a colletion of Network Service Clients, Network Service Endpoints, and the Network Service Dataplane for those NSCs and NSEs within its Network Service Manager Domain.  An example of a Network Service Manager Domain would be a single Kubernetes Node.  A Network Service Manager is responsible to
+        <ul> 
+            <li>Advertise Network Service Endpoints in its domain to zero or more Network Service Registries</li>
+            <li>Establish L2/L3 Connections directly between Network Service Clients in its domain and Network Service Endpoints in its domain.
+            <li>Collaborate with other Network Service Managers discovered via one or more Network Service Registries to establish L2/L3 Connections between Network Service Clients in its domain and Network Service Clients in another Network Service Manager's domain.</li>
+            <li>Collaborate with other Network Service Managers to establish L2/L3 Connections between Network Service Endpoints in its domain and Network Service Clients in abother Network Service Manager's domain.
+        </ul>
+    </dd>
+    <dt>Network Service Registry (NSR)</dt>
+    <dd>
+        A Network Service Registry is used to register:
+        <ul> 
+            <li>Network Services</li>
+            <li>Network Service Endpoints</li>
+        </ul>
+        A Network Service Registry allows Network Service Managers to advertise and find each others Network Service Endpoints within its Network Service Registy Domain.
+        An example of a Network Service Registry would be Network Service CRDs and Network Service Endpoint CRDs stored in the Kubernetes API server for a Kubernetes Cluster.
```

Comment:
CRD is just a schema definition of Custom Resource (CR), people often use them interchangeably but it is not correct. NSM advertises Custom Resources. Also I thought Service Registry would be a k8s cluster which is storing NS and NSE CRs.

Reply:
I too am slightly confused by this NSR concept. I agree with @sbezverk's comment about the k8s cluster storing the NS and NSE CRs.

Reply:
CRD -> CR should be fixed in latest push on PR.